pyccel/codegen/printing/fcode.py

```diff
@@ -2653,16 +2653,9 @@ def _print_Header(self, expr):
         return ''
 
     def _print_ConstructorCall(self, expr):
-        func = expr.func
-        name = str(func.name)
-        if name == "__init__":
-            name = "create"
-        name = self._print(name)
-
-        code_args = ''
-        if expr.arguments is not None:
-            code_args = ', '.join(self._print(i) for i in expr.arguments)
-        code = '{0}({1})'.format(name, code_args)
+        code = self._print_FunctionCall(expr)
+        for k, m in list(_default_methods.items()):
+                code = code.replace(k, m)
```

Comment:
I think it would be simpler to put this code in `_print_FunctionCall` so we know the name has been correctly changed at every call. If you want to keep it here then there is no need to use `_default_methods`.

By the way, it is probably a bad idea to cast to a list here. It is unnecessary, the following code will run:
```suggestion
        for k, m in _default_methods.items():
                code = code.replace(k, m)
```
But worse casting to a list means that a list is created in memory so all the elements are copied and saved, without the list this extra memory is not needed